bitnami/contour/templates/default-backend/ingress.yaml

```diff
@@ -19,7 +19,15 @@ metadata:
     {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
     {{- end }}
 spec:
+  {{- if semverCompare "<1.19-0" (include "common.capabilities.kubeVersion" .) -}}
   backend:
     serviceName: {{ include "common.names.fullname" . }}-default-backend
     servicePort: http
```

Comment:
Hi,

Would this work without the `rules` section in old k8s versions?

Reply:
Jup, this works.
And here's the doc ref:

# 1.20

Field | Description
-- | --
default _BackendIngressBackend_ | DefaultBackend is the backend that should handle requests that don't match any rule. If Rules are not specified, DefaultBackend must be specified. If DefaultBackend is not set, the handling of requests that do not match any of the rules will be up to the Ingress controller.
rules _IngressRule array_ | A list of host rules used to configure the Ingress. If unspecified, or no rule matches, all traffic is sent to the default backend.

Ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#ingressspec-v1-networking-k8s-io

# 1.18

Field | Description
-- | --
backend _IngressBackend_ | A default backend capable of servicing requests that don't match any rule. At least one of 'backend' or 'rules' must be specified. This field is optional to allow the loadbalancer controller or defaulting logic to specify a global default.
rules _IngressRule array_ | A list of host rules used to configure the Ingress. If unspecified, or no rule matches, all traffic is sent to the default backend.

Ref: https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#ingressspec-v1beta1-networking-k8s-io

# 1.16

Field | Description
-- | --
backend _IngressBackend_ | A default backend capable of servicing requests that don't match any rule. At least one of 'backend' or 'rules' must be specified. This field is optional to allow the loadbalancer controller or defaulting logic to specify a global default.
rules _IngressRule array_ | A list of host rules used to configure the Ingress. If unspecified, or no rule matches, all traffic is sent to the default backend.

Ref: https://v1-16.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#ingressspec-v1beta1-networking-k8s-io